src/caffe/layers/sigmoid_cross_entropy_loss_layer.cu

```diff
@@ -11,21 +11,9 @@ namespace caffe {
 template <typename Dtype>
 void SigmoidCrossEntropyLossLayer<Dtype>::Forward_gpu(
     const vector<Blob<Dtype>*>& bottom, const vector<Blob<Dtype>*>& top) {
-  // The forward pass computes the sigmoid outputs.
-  sigmoid_bottom_vec_[0] = bottom[0];
-  sigmoid_layer_->Forward(sigmoid_bottom_vec_, sigmoid_top_vec_);
```

Comment:
That might have been what I was thinking, but `Forward` will check `Caffe::mode()` and decide whether to call `Forward_cpu` or `gpu` based on that.